src/containers/tud/components/QuestionnaireForm.js

```diff
@@ -58,6 +58,50 @@ const ButtonRow = styled.div`
   margin-top: 30px;
 `;
 
+/*
+ * This code fixes an issue where the survey enters an error state after a user fills out the survey
+ * and later changes the day end time value. If the day end time is changed such that certain sections
+ * of the formRef state contain invalid data (the start/end values are out of range), then this will cause
+ * validateAndSubmit to fail, hence the need to remove such sections. This deletion need only take place
+ * just before we enter the nightActivity portion of the survey. At this point, for all x s.t x > page
+ * only x == page + 1 can be expected to contain any data
+  (and it should be nightActivity data otherwise it also needs to be deleted)
+ */
+
+const removeExtraData = (formRef :Object, pagedData, page :number) => {
+  const psk = getPageSectionKey(page, 0);
+  const dayEndTime :?DateTime = selectTimeByPageAndKey(DAY_SPAN_PAGE, DAY_END_TIME, pagedData);
+  const formData :Object = formRef?.current?.state?.formData || {};
+
+  const currEndTime = formData[psk]?.[ACTIVITY_END_TIME];
+
+  if (dayEndTime && currEndTime) {
+    const currEndDateTime :DateTime = DateTime.fromSQL(currEndTime);
+    if (!currEndDateTime.equals(dayEndTime)) {
+      return;
+    }
+    const pages = Object.keys(formData)
+      .map((key) => parsePageSectionKey(key))
+      .map((parsedResult) => parsedResult.page)
+      .map((pageStr) => Number(pageStr));
```

Comment:
You can do this in 1 `.map()`, why did you go with 3 `.map()`?

```
const pages = Object.keys(formData)
  .map((key :string) => {
    const psk = parsePageSectionKey(key);
    return Number(psk.page);
  });
```

Reply:
the suggested has arguably better performance, so will switch.